roles/ec2-pod/tasks/main.yml

```diff
@@ -0,0 +1,133 @@
+---
+
+  - name: variables | control | compute
+    set_fact:
+        do_add: >
+            {{ state == "running" }}
+        do_create: >
+            {{ state == "running" or state == "stopped" }}
+        do_destroy: >
+            {{ state == "absent" }}
+        do_wait: >
+            {{ state == "running" }}
+
+  - name: hosts spec | process
+    set_fact:
+        hosts_spec: "{{ hosts|process_hosts_spec(name) }}"
+
+  - name: instances | update lists | compute
+    set_fact:
+        update_lists: >
+            {{ name|compute_ec2_update_lists(hosts,
+                                             state,
+                                             region,
+                                             default_ssh_key,
+                                             default_image,
+                                             default_instance_type) }}
+
+  - name: instances | terminate set | compute
+    set_fact:
+        terminate_set: >
+            {{ (update_lists.terminate
+                if (update_lists.terminate|length > 0)
+                else [])|union(update_lists.start
+                               if ((do_destroy|bool) and
+                                   (update_lists.start|length) > 0)
+                               else []) }}
+
+  - name: security group | create
+    ec2_group:
+        name: ec2_pod_{{ name }}
+        description: >
+            security group for ec2 pod: {{ name }}
+        region: "{{ region }}"
+        rules: "{{ rules }}"
+        rules_egress: "{{ rules_egress }}"
+    when: do_create|bool
+
+  - name: instances | terminate set | terminate
+    ec2:
+        instance_ids: "{{ terminate_set }}"
+        region: "{{ region }}"
+        state: absent
+        wait: yes
+        wait_timeout: 120 # two minutes
+    ignore_errors: true
+    when: (terminate_set|length > 0)|bool
+
+  - name: instances | start set | start
+    ec2:
+        instance_ids: "{{ update_lists.start }}"
+        region: "{{ region }}"
+        state: running
+        wait: yes
+        wait_timeout: "{{ EC2_TIMEOUT }}"
+    when: ((do_create|bool) and (update_lists.start|length) > 0)|bool
+
+  - name: instances | create
+    ec2:
+        count_tag:
+            Name: >
+                {{ item.value.name|default(
+                    ["ec2_pod", name, item.key]|join("_")) }}
+            ec2_pod: "{{ name }}"
+            ec2_pod_instance_name: "{{ item.key }}"
+        exact_count: "{{ (item.value.count|default(1))|int }}"
+        group: ec2_pod_{{ name }}
+        image: "{{ item.value.image|default(default_image) }}"
+        instance_tags:
+            Name: >
+                {{ item.value.name|default(
+                    ["ec2_pod", name, item.key]|join("_")) }}
+            ec2_pod: "{{ name }}"
+            ec2_pod_instance_name: "{{ item.key }}"
```

Comment:
@opadron It would be nice if we could get tags here that marked what group each host was in.  I'm thinking in particular if you've got 20 or 30 machines running it would be nice to be able to run something like:

```bash
#/bin/bash

IP=$(aws ec2 describe-instances --filters "Name=girder" \
     --query Reservations[0].Instances[0].PublicIpAddres)

ssh -o StrictHostKeyCHecking=no -i /path/to/pem/file.pem ubuntu@$IP
```

to pop into the instance running girder without having to pick through the management console to find the right machine.